lib/bib.rb

```diff
@@ -0,0 +1,29 @@
+require 'httparty'
+
+class Bib < MarcRecord
+  def is_research?
+    data = get_platform_api_data
```

Comment:
Given how `Item` is implemented (and their common base class) I expected `get_platform_api_data` to return the marc data for the bib itself. I was surprised to discover later that `data` is actually an array of descendant items. The  reuse of the super `get_platform_api_data` with class-specific `api_path` method is nice, but I think it implies a symmetry that is misleading.

I think this could be resolved by passing the path directly to `get_platform_api_data` like:

```
items = get_platform_api_data items_api_path

result = @is_partner || has_zero_items?(items) || has_at_least_one_research_item?(items)
```

Also, what do you think of just using `is_partner?` instead of `@is_partner` so that you don't have to call `get_platform_api_data` and can skip an unnecessary fetch if `is_partner? == true`.. Something like

```
def is_research?
  if is_partner?
    result = true
  else
    items = get_platform_api_data items_api_path
    result = has_zero_items?(items) || has_at_least_one_research_item?(items)
  end

  $logger.debug "Evaluating is-research for #{nypl_source} #{id}: #{result}", @log_data

  result
end
```

I don't know; Maybe that complexity isn't worth the savings.

If you wanted to keep the logic on one-line, you could implement the `has_` methods to fetch data on-demand:

```
def is_research?
  result = is_partner? || has_zero_items? || has_at_least_one_research_item?

  $logger.debug "Evaluating is-research for #{nypl_source} #{id}: #{result}", @log_data

  result
end

def has_zero_items?
  items.size == 0
end

def items
  @items ||= get_platform_api_data items_api_path
end

...
```

Reply:
Mentioned in a previous comment as well, only need to fetch data to validate that the record does indeed exist. Otherwise, I agree with all the suggestions!